js/ui/appDisplay.js

```diff
@@ -1519,6 +1519,11 @@ const AppIcon = new Lang.Class({
                 this._updateRunningStyle();
             }));
         this._updateRunningStyle();
+
+        if (app.get_id() === 'com.endlessm.Coding.Chatbox.desktop') {
+            Main.notificationDaemon.gtk.connect('new-gtk-notification-source',
+                                                Lang.bind(this, this._onNewGtkNotificationSource));
+        }
```

Comment:
Why would we want to only do this for the chatbox? Is it to prevent "counters" on other desktop icons?
Also since the notification daemon will outlive the app icon, if we implement it like this we'll have to be careful to disconnect this signal when the icon is destroyed.